[RHDHBUGS-2481]: Add LDAP UUID sign-in resolver setup instructions (#2042)
redhat-developer/red-hat-developers-documentation-rhdh: modules/shared/proc-configure-the-ldap-uuid-sign-in-resolver-for-rhbk.adoc

:_mod-docs-content-type: PROCEDURE

[id="configure-the-ldap-uuid-sign-in-resolver-for-rhbk_{context}"]
= Configure the LDAP UUID sign-in resolver for {rhbk-brand-name}

[role="_abstract"]

> When you use {rhbk-brand-name} with LDAP user federation, you can configure the `oidcLdapUuidMatchingAnnotation` sign-in resolver to match users by their immutable LDAP UUID.

Reviewer: ```suggestion
When you use {rhbk-brand-name} with LDAP user federation, configure the `oidcLdapUuidMatchingAnnotation` sign-in resolver to match users by their immutable LDAP UUID for secure user resolution.
```

The idea is to make it clear that for production environments, we should recommend setting up this resolver for best security practices.

This requires a custom client scope in {rhbk-brand-name} that exposes the LDAP UUID as a token claim.

.Prerequisites

* LDAP user federation is configured in {rhbk-brand-name}.
* LDAP provisioning is enabled in {product}.

> For more information, see xref:enable-user-provisioning-with-ldap_{context}[Enable user provisioning with LDAP].

Reviewer 2: (Should this be in one line?)

* A {product} client is created in {rhbk-brand-name}.

.Procedure

. In the {rhbk-brand-name} admin console, go to *Client scopes* and click *Create client scope*.
Name the scope `ldap_uuid`.

. In the `ldap_uuid` scope, click the *Mappers* tab, then click *Add mapper* > *Configure a new mapper* > *User Attribute*.
Configure the mapper with the following values:
+
--
* *Name*: `LDAP_ID`
* *User Attribute*: `LDAP_ID`
* *Token Claim Name*: `ldap_uuid`
* *Claim JSON Type*: `String`
* *Add to ID token*: *ON*
* *Add to userinfo*: *ON*
--

. Go to *Clients* > your {product} client > *Client scopes*.

> Click *Add client scope* and add `ldap_uuid` as a *Default* scope.

Reviewer 2: (I think this should be a separate step?)

> . Add the `oidcLdapUuidMatchingAnnotation` resolver to your `{my-app-config-file}` file:

Reviewer: This snippet is not mandatory. If `ldapUuidKey` is not configured, it'll default to `ldap_uuid`

Author: ```suggestion
. Optionally, add the `oidcLdapUuidMatchingAnnotation` resolver to your `{my-app-config-file}` file, to replace the default `ldap_uuid` resolver:
```

> . Optionally, add the `oidcLdapUuidMatchingAnnotation` resolver to your `{my-app-config-file}` file, to replace the default `ldap_uuid` resolver:

Reviewer 2: ```suggestion
. Optional: Add the `oidcLdapUuidMatchingAnnotation` resolver to your `{my-app-config-file}` file, to replace the default `ldap_uuid` resolver:
```
